prometheus-exporter/src/main/java/be/xplore/githubmetrics/prometheusexporter/workflowrun/WorkflowRunBuildTimesOfLastDayExporter.java

```diff
@@ -0,0 +1,134 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WorkflowRunBuildTimesOfLastDayExporter.class);
+    private final GetAllWorkflowRunBuildTimesOfLastDayUseCase getAllWorkflowRunBuildTimesOfLastDayUseCase;
+    private final MeterRegistry registry;
+    private final String cronExpression;
+
+    public WorkflowRunBuildTimesOfLastDayExporter(
+            GetAllWorkflowRunBuildTimesOfLastDayUseCase
+                    getAllWorkflowRunBuildTimesOfLastDayUseCase,
+            SchedulingProperties schedulingProperties,
+            MeterRegistry registry
+    ) {
+        this.getAllWorkflowRunBuildTimesOfLastDayUseCase
+                = getAllWorkflowRunBuildTimesOfLastDayUseCase;
+        this.registry = registry;
+        this.cronExpression = schedulingProperties.workflowRunBuildTimesInterval();
+    }
+
+    private void retrieveAndExportLastDaysWorkflowRunBuildTimes() {
+        LOGGER.info("Running scheduled workflow runs build times task");
+        List<WorkflowRun> workflowRuns =
+                getAllWorkflowRunBuildTimesOfLastDayUseCase.getAllWorkflowRunBuildTime();
+
+        var buildTimesMap = new HashMap<String, Map<WorkflowRunStatus, Integer>>();
+
+        buildTimesMap.put("workflow_runs_total_build_times", this.getTotalBuildTimes(workflowRuns));
+        buildTimesMap.put("workflow_runs_average_build_times", this.getAverageBuildTimes(workflowRuns));
+
+        this.publishWorkflowRunsBuildTimesGauges(buildTimesMap);
+    }
+
+
+    private void publishWorkflowRunsBuildTimesGauges(
+            Map<String, Map<WorkflowRunStatus, Integer>> buildTimesMap
+    ) {
+        for (Map.Entry<String, Map<WorkflowRunStatus, Integer>> metric : buildTimesMap.entrySet()) {
+            for (Map.Entry<WorkflowRunStatus, Integer> entry : metric.getValue().entrySet()) {
+                Gauge.builder(metric.getKey(),
+                                entry,
+                                Map.Entry::getValue
+                        ).tag("status", entry.getKey().toString())
+                        .strongReference(true)
+                        .register(this.registry);
+
+            }
+
+        }
+    }
+
+    private Map<WorkflowRunStatus, Integer> getAverageBuildTimes(List<WorkflowRun> workflowRuns) {
+        Map<WorkflowRunStatus, Integer> workflowRunStatusCountMap
+                = this.createExportMap();
+        Map<WorkflowRunStatus, Integer> workflowRunExportMap
+                = this.createExportMap();
+
+        workflowRuns.forEach(workflowRun -> {
+            workflowRunStatusCountMap.put(
+                    workflowRun.getStatus(),
+                    1 + workflowRunStatusCountMap.get(
+                            workflowRun.getStatus()));
+            workflowRunExportMap.put(
+                    workflowRun.getStatus(),
+                    workflowRun.getBuildTime() +
+                            workflowRunExportMap.get(workflowRun.getStatus())
+            );
+        });
```

Comment:
I wonder if you could actually collect all the build times into a list of build times per status because that could then easily be summed and devided by the length of the list at the same time. Meaning you would have `Map<Status, List<float>>` and then you could do for every list `list.sum()/list.size()`. Tell me what you think

Reply:
fixed